pkg/pillar/utils/generics/lockmap.go

```diff
@@ -75,3 +75,20 @@ func (lm *LockedMap[K, V]) Range(callback LockedMapFunc[K, V]) {
 	}
 	lm.RUnlock()
 }
+
+// LockedMapApplyFunc is function signature for ApplyOrStore function
+type LockedMapApplyFunc[V any] func(V) V
+
+// ApplyOrStore applies to callback and store new value or store
+// value provided if key is not stored in map. This funcion is actually
+// useful when you want to implement atomic counter
```

Comment:
I'm having a hard time reading the comment even though I know what the function does.  How about something like
"ApplyOrStore is used to perform atomic operations, where the operation is performed by the callback function. If the key does not exist in the map then it will be created with the initial values set by the callback function. Otherwise the entry for the key will be updated."

Also, it might be more clear to rename "callback" to something like "updatefn"/"updateFunction" or applyFn as you have in the type definition?

It might be that future uses would want to know whether the key existed so the function can do different things for new vs. existing entries. But I don't know if it makes sense to add some "bool existing" to the LockedMapApplyFunc argument list. Probably overkill to do that now.

Reply:
Changed comment but tweaked one line from your suggestion: if it doesn't exist we simply store default value, applyFn doesn't create default values. And also added bool to function signature, because it's useful, although not needed in this particular case

Reply:
> Changed comment but tweaked one line from your suggestion: if it doesn't exist we simply store default value, applyFn doesn't create default values. And also added bool to function signature, because it's useful, although not needed in this particular case

OK but if applyfn does an increment of some field, then that field will start with a value of 1. Does the comment capture that?